docs/configuration/configuration.md

```diff
@@ -1021,7 +1021,9 @@ Additional labels prefixed with `__meta_` may be available during the
 relabeling phase. They are set by the service discovery mechanism that provided
 the target and vary between mechanisms.
 
-Labels starting with `__` will be removed from the label set after relabeling is completed.
+Labels starting with `__` will be removed from the label set after configuration
+relabeling is completed. With alert and metric relabeling, it is required
```

Comment:
This is the section on relabel_configs, and you also missed write relabelling. I'd suggest limiting this to mention that this is specific to target relabelling.